File: backend/schemas/prediction.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class PredictionCreate(BaseModel):
    match_id: int
    goles_local_pred: int
    goles_visita_pred: int
    goleador_pred: Optional[str] = None
    jugador_tarjeta_pred: Optional[str] = None
    tipo_tarjeta_pred: Optional[str] = None
    minuto_gol_pred: Optional[int] = None

class PredictionUpdate(BaseModel):
    goles_local_pred: Optional[int] = None
    goles_visita_pred: Optional[int] = None
    goleador_pred: Optional[str] = None
    jugador_tarjeta_pred: Optional[str] = None
    tipo_tarjeta_pred: Optional[str] = None
    minuto_gol_pred: Optional[int] = None

class PredictionResponse(BaseModel):
    id: int
    user_id: int
    match_id: int
    goles_local_pred: int
    goles_visita_pred: int
    goleador_pred: Optional[str] = None
    jugador_tarjeta_pred: Optional[str] = None
    tipo_tarjeta_pred: Optional[str] = None
    minuto_gol_pred: Optional[int] = None
    fecha_prediccion: datetime

    class Config:
        from_attributes = True

class PredictionPublicResponse(BaseModel):
    id: int
    user_id: int
    user_nombre: str
    match_id: int
    goles_local_pred: int
    goles_visita_pred: int
    goleador_pred: Optional[str] = None
    jugador_tarjeta_pred: Optional[str] = None
    tipo_tarjeta_pred: Optional[str] = None
    minuto_gol_pred: Optional[int] = None
    fecha_prediccion: datetime

File: predix-frontend/types/prediction.ts
export interface Prediction {
  id: number;
  user_id: number;
  match_id: number;
  goles_local_pred: number;
  goles_visita_pred: number;
  goleador_pred: string | null;
  jugador_tarjeta_pred: string | null;
  tipo_tarjeta_pred: string | null;
  minuto_gol_pred: number | null;
  fecha_prediccion: string;
}

export interface PredictionPublic extends Prediction {
  user_nombre: string;
}

export interface CreatePredictionPayload {
  match_id: number;
  goles_local_pred: number;
  goles_visita_pred: number;
  goleador_pred?: string;
  jugador_tarjeta_pred?: string;
  tipo_tarjeta_pred?: string;
  minuto_gol_pred?: number;
}

export interface UpdatePredictionPayload {
  goles_local_pred?: number;
  goles_visita_pred?: number;
  goleador_pred?: string;
  jugador_tarjeta_pred?: string;
  tipo_tarjeta_pred?: string;
  minuto_gol_pred?: number;
}
